internal/network/connectivity_groups.go

```diff
@@ -482,7 +482,13 @@ func newL3QueryFactory(hosts []*models.Host, family AddressFamily) (hostQueryFac
 				if err != nil {
 					return nil, err
 				}
-				value[ip.String()] = true
+				ipInCidr, err := IpInCidr(ip.String(), primaryMachineCIDR)
```

Comment:
This misses the whole point of creating connectivity-groups.  The connectivity-groups should provide all connectivity information.  Then the validation should filter weather it needs L2, L3 or specific networks.

Reply:
I see...I couldn't find another place to filter ip addresses 😅 do you know of another place where we can filter them?

Reply:
Actually you are right.  If we want to limit the addresses this is probably the place.
In general, in UMN any IP from any network can be part of the cluster.  Therefore, no machine-networks are required for multi-node UMN.
Since no machine-network should be provided in the install-config, OCP doesn't expect them either.
So even if we limit the addresses we use in connectivity-check, does OCP limit its activity to these addresses?

Reply:
Hmm not too sure, @zaneb do you know if OCP will use these addresses? Also if no machine networks are required for UMN then how do we filter them?

Reply:
OCP doesn't really seem to use the machine-networks except for validations in the installer.

In UMN hosts are allowed to be in different networks. Because assisted has a limitation of only 1 machine-network, I guess we just allow none to be specified. #6071 removes that limitation. We still don't want any of the machine networks to overlap with e.g. the clusterNetwork or serviceNetwork, so this enables us to do all of the validations.

In any event, the agent installer always passes the machineNetwork(s), even with UMN.